docs/concepts/api-extension/custom-resources.md

```diff
@@ -42,12 +43,162 @@ The [Operator](https://coreos.com/blog/introducing-operators.html) pattern is on
 combination. It allows developers to encode domain knowledge for specific applications into an
 extension of the Kubernetes API.
 
+### Should I Add a Custom Resource to my Kubernetes Cluster?
+When creating a new API, first consider if you should add your API to Kubernetes cluster APIs, or instead let your API stand alone.
+
+
+<table>
+  <tr>
+   <td>Consider adding an API to the Kubernetes API if...
+   </td>
+   <td>Prefer a stand-alone API if...
+   </td>
+  </tr>
+  <tr>
+   <td>… your API is Declarative. Typically, this means:<ul>
+<li>your API consists of a relatively small number of relatively small objects
+<li>the objects define configuration of applications or infrastructure
+<li>the objects are updated relatively infrequently
+<li>humans often need to read and write the objects
+<li>the main operations on the objects are (CRUD) creating, reading, updating and deleting
+<li>humans often need to read and write the objects
+<li>transactions across objects are not required.</li></ul>
+   </td>
+   <td>… your API does not fit the Declarative model.  For example, it might:<ul>
+<li>need to directly store large amounts of data (e.g. > a few kB per object, or >1000s of objects)
+<li>need to be high bandwidth (10s of requests per second sustained)
+<li>store end-user data (such as images, PII, etc) or other large-scale data processed by applications<li>
+<li>the objects require non-CRUD operations.
+<li>the API does is not well modeled as objects.</li></ul>
+   </td>
+  </tr>
+  <tr>
+   <td>...you want your new types to be readable and writable using the kubernetes CLI.  For example:<ul>
```

Comment:
Another way of putting this is: you want to manage config for your new types the same way you manage config for Kubernetes resources like Deployments, ConfigMaps, etc. For example, maybe you put all your k8s manifests in source control and you never touch kubectl because CI/CD transparently applies them for you. A custom resource API makes sense if you want to put the config for the new API in the same place.